src/libraries/Microsoft.Extensions.Logging.Abstractions/gen/LoggerMessageGenerator.Emitter.cs

```diff
@@ -60,7 +60,8 @@ private static bool UseLoggerMessageDefine(LoggerMethod lm)
                     for (int i = 0; i < lm.TemplateList.Count; i++)
                     {
                         string t = lm.TemplateList[i];
-                        if (!t.Equals(lm.TemplateParameters[i].Name, StringComparison.OrdinalIgnoreCase))
+                        var (template, parameter) = SanitizeAtSign(t, lm.TemplateParameters[i].CodeName);
```

Comment:
nit: 
Can we use the span here instead of string? I mean having

```C#
        static ReadOnlySpan<char> SanitizeAtSign(ReadOnlySpan<char> template)
            => (template.Length > 0 && template[1] == '@') ? template.Slice(1) : template;
```

and do 

```
    if (!SanitizeAtSign(template).Equals(SanitizeAtSign((parameter), StringComparison.OrdinalIgnoreCase))
       ...
```

Reply:
@tarekgh just uploaded code to make use of ReadOnlySpan. Also renamed method so we can handle '$' symbols in variable too.